airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py

```diff
@@ -107,7 +107,57 @@ def read_records(
     ) -> Iterable[Mapping[str, Any]]:
         """Main read method used by CDK"""
         for record in self._read_records(params=self.request_params(stream_state=stream_state)):
-            yield self._extend_record(record, fields=self.fields)
+            yield self.transform(self._extend_record(record, fields=self.fields))
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Use this method to remove update fields types in record according to schema.
+        """
+        schema = self.get_json_schema()
+        self.convert_to_schema_types(record, schema["properties"])
+        return record
+
+    def get_python_type(self, _types: Union[list, str]) -> tuple:
+        types_mapping = {
+            "string": str,
+            "number": float,
+            "integer": int,
+            "null": None,
+            "object": dict,
+            "array": list,
+            "boolean": bool,
+        }
+
+        if isinstance(_types, list):
+            return tuple([types_mapping[t] for t in _types if t != "null"])
+
+        return (types_mapping[_types],)
+
+    def convert_to_schema_types(self, record: Mapping[str, Any], schema: Mapping[str, Any]):
+        """
+        Converts values' type from record to appropriate type from schema. For example, let's say we have `reach` value
+        and in schema it has `number` type because it's, well, a number, but from API we are getting `reach` as string.
+        This function fixes this and converts `reach` value from `string` to `number`. Same for all fields and all
+        types from schema.
+        """
+        for key, value in record.items():
+            if key not in schema:
+                continue
+
+            if isinstance(value, dict):
+                self.convert_to_schema_types(record=value, schema=schema[key].get("properties", {}))
+            elif isinstance(value, list) and "items" in schema[key]:
+                for record_list_item in value:
+                    if list in self.get_python_type(schema[key]["items"]["type"]):
+                        # TODO Currently we don't have support for list of lists.
+                        pass
+                    elif dict in self.get_python_type(schema[key]["items"]["type"]):
```

Comment:
the expression `schema[key][items][type]` is repeated 3 times, might be good to refactor into a variable for readability e.g: `list_items_type = schema[key][items][type]`

Reply:
Updated.

Reply:
Nope, we can't do that. `schema[key]["items"]["type"]` is being used specifically in `if` statements in which we are sure key `items` is presented. If I will move `schema[key]["items"]["type"]` in one place above all `if` statements then we will keep on getting errors like [this](https://github.com/airbytehq/airbyte/runs/3596290846?check_suite_focus=true) one.